In [1]:
import numpy as np
import pandas as pd
import pickle
import json
import random
import codecs

# extract the information needed from the big json file  


I didn't upload the original tweets file,bucause it's too huge,and I extracted only the information I need into another json file,called "tweets1" and "tweets2"

In [30]:

f = open("sample_2.json",'r',encoding='utf-8')
ln = 0
for line in f.readlines():
    ln += 1
    dic = json.loads(line)
    with codecs.open('tweets2.json','a', 'utf-8') as outf:
        
        if 'media' in dic['entities']:
            json.dump({'id': dic['_id'], 'hashtags':dic['entities']['hashtags'] ,'urls':dic['entities']['urls'],'media':dic['entities']['media']},outf, ensure_ascii=False)
        else:
            json.dump({'id': dic['_id'], 'hashtags':dic['entities']['hashtags'] ,'urls':dic['entities']['urls'],'media':[]},outf, ensure_ascii=False)
        outf.write('\n')
f.close()

In [ ]:
f1 = open("sample_1.json",'r',encoding='utf-8')
ln1 = 0
for line in f1.readlines():
    ln1 += 1
    dic = json.loads(line)
    with codecs.open('tweets1.json','a', 'utf-8') as outf:
        
        if 'media' in dic['entities']:
            json.dump({'id': dic['_id'], 'hashtags':dic['entities']['hashtags'] ,'urls':dic['entities']['urls'],'media':dic['entities']['media']},outf, ensure_ascii=False)
        else:
            json.dump({'id': dic['_id'], 'hashtags':dic['entities']['hashtags'] ,'urls':dic['entities']['urls'],'media':[]},outf, ensure_ascii=False)
        outf.write('\n')
f1.close()

## For sample_1 

In [296]:

f = open("tweets1.json",'r',encoding='utf-8')
ln = 0
sp1_urls = 0
sp1_media = 0
sp1_hash1 = 0
sp1_hash2 = 0
sp1_hashlist1 = []
sp1_hashlist2 = []


for line in f.readlines():
    ln += 1
    hlist = []
    dic = json.loads(line)
    
    if dic['urls'] != []:
        sp1_urls += 1
    if dic['media'] != []:
        sp1_media += 1
    if dic['hashtags'] !=[]:
        sp1_hash1 += 1
        for i in dic['hashtags']:
            sp1_hashlist1.append(i['text'])
        if len(dic['hashtags']) >= 2:
            #print(dic['id'])
            for j in dic['hashtags']:
                hlist.append(j['text'])
            sp1_hashlist2.append(hlist)


### Hashtags,Urls,Media

In [384]:
print("Total number of tweets in sample_1: ",ln )
print("Total number of tweets contain urls in sample_1: ",sp1_urls,"  pertcentage: ",sp1_urls/ln  )
print("Total number of tweets contain media in sample_1: ",sp1_media,"  pertcentage: ",sp1_media/ln  )
print("Total number of tweets contain at least 1 hashtag in sample_1: ",sp1_hash1,"  pertcentage: ",sp1_hash1/ln  )
print("Total number of tweets contain more than 2 hashtags in sample_1: ",len(sp1_hashlist2),"  pertcentage: ",len(sp1_hashlist2)/ln  )

Total number of tweets in sample_1:  30000
Total number of tweets contain urls in sample_1:  6911   pertcentage:  0.23036666666666666
Total number of tweets contain media in sample_1:  5138   pertcentage:  0.17126666666666668
Total number of tweets contain at least 1 hashtag in sample_1:  7224   pertcentage:  0.2408
Total number of tweets contain more than 2 hashtags in sample_1:  3299   pertcentage:  0.10996666666666667


Total number of tweets in sample_1:  30000

Total number of tweets contain urls in sample_1:  6911   pertcentage:  0.23036666666666666

Total number of tweets contain media in sample_1:  5138   pertcentage:  0.17126666666666668

Total number of tweets contain at least 1 hashtag in sample_1:  7224   pertcentage:  0.2408

Total number of tweets contain more than 2 hashtags in sample_1:  3299   pertcentage:  0.10996666666666667

### The 50 most frequent hashtags in sample_1:

In [423]:
result = Counter(sp1_hashlist1)
d = sorted(result.items(), key=lambda x: x[1], reverse=True)
df1_1 = pd.DataFrame(d,columns = ["hashtags ","numbers"]).head(50)

In [424]:

ranks = pd.DataFrame(list(range(1,51)),columns=["Rank"]).pop('Rank')
df1_1.insert(0,'Rank',ranks)

This is the most 50 most frequent hashtags in sample_1 and its rank:

In [445]:
df1_1

Rank             hashtags   numbers
0      1           switzerland       98
1      2        CanYouMakeIt16       81
2      3                   art       70
3      4           Switzerland       55
4      5                  quiz       53
5      6                Hiring       53
6      7        noinonmolliamo       52
7      8         ODproroga2018       51
8      9                   Job       46
9     10                  Jobs       44
10    11                Geneva       44
11    12               NANTAIS       43
12    13  INCLingapSaTondo2016       41
13    14                   job       40
14    15              TheVoice       40
15    16           Controleurs       39
16    17             CareerArc       39
17    18                zurich       38
18    19                NANTES       37
19    20                  TPMP       37
20    21                   CFP       33
21    22         GreenUrbanism       33
22    23              Abstract       33
23    24              Deadline       33
24    25                  May1       33
25    26                  BATB       32
26    27                travel       31
27    28                  lake       31
28    29          bahnbilderde       28
29    30           photography       28
30    31           askbelieber       28
31    32                 Basel       27
32    33                Zurich       27
33    34              Veterans       26
34    35               TDR2016       26
35    36                  como       25
36    37                spring       25
37    38                  love       24
38    39                nature       24
39    40            TPMPAVegas       24
40    41           farmaciauno       23
41    42                sconti       23
42    43                 ggrmc       22
43    44            Radolfzell       22
44    45                 swiss       22
45    46                   LHC       21
46    47                hiring       21
47    48                 Sales       21
48    49                design       21
49    50             LesAnges8       21

### The 50 most frequent hashtags pairs in sample_1

In [340]:
pairs_s1 =[]


for i in sp1_hashlist2:
    
    for m in i:
        for j in i:
            templist = []
            if j != m:
                templist.append(j)
                templist.append(m)
                pairs_s1.append(templist)

In [356]:
pairlist_s1 = []
for i in pairs_s1:
    if (i not in pairlist_s1) & (i[::-1] not in pairlist_s1):
        pairlist_s1.append(i)

In [366]:
count_pair1 =[]
for i in pairlist_s1:
    templist = []
    templist.append(i)
    templist.append(pairs_s1.count(i))
    count_pair1.append(templist)

In [455]:
columnsh = ["hashtags ","numbers"]
count_pdf1 = pd.DataFrame(count_pair1,columns = columnsh)
df1_2 = count_pdf1.sort_values(by="numbers" , ascending=False).head(50)
df1_2.index=[i for i in range(df1_2.shape[0])]

In [456]:

ranks = pd.DataFrame(list(range(1,51)),columns=["Rank"]).pop('Rank')
df1_2.insert(0,'Rank',ranks)

The 50 most frequent hashtags pairs and their ranks in sample_1


In [457]:
df1_2

Rank                        hashtags   numbers
0      1  [noinonmolliamo, ODproroga2018]       51
1      2                      [Jobs, Job]       42
2      3        [Deadline, GreenUrbanism]       33
3      4                 [May1, Deadline]       33
4      5                 [May1, Abstract]       33
5      6             [GreenUrbanism, CFP]       33
6      7            [May1, GreenUrbanism]       33
7      8                  [Abstract, CFP]       33
8      9        [Abstract, GreenUrbanism]       33
9     10                      [May1, CFP]       33
10    11                  [Deadline, CFP]       33
11    12             [Deadline, Abstract]       33
12    13              [CareerArc, Hiring]       30
13    14                    [Hiring, job]       28
14    15                    [Hiring, Job]       25
15    16                   [Hiring, Jobs]       25
16    17            [sconti, farmaciauno]       23
17    18                 [CareerArc, Job]       22
18    19                [CareerArc, Jobs]       22
19    20             [promozioni, sconti]       21
20    21        [promozioni, farmaciauno]       21
21    22              [donna, promozioni]       20
22    23                  [donna, sconti]       20
23    24             [donna, farmaciauno]       20
24    25                  [painting, art]       20
25    26                [bellezza, donna]       18
26    27           [bellezza, promozioni]       18
27    28                  [iloveart, art]       18
28    29               [bellezza, sconti]       18
29    30          [bellezza, farmaciauno]       18
30    31          [endorphins, Endomondo]       17
31    32               [Hiring, Veterans]       17
32    33                 [CareerArc, job]       17
33    34                   [fineart, art]       17
34    35                    [artwit, art]       16
35    36           [Veterans, Radolfzell]       14
36    37                  [Veterans, Job]       14
37    38              [iloveart, fineart]       14
38    39               [artwit, painting]       14
39    40                 [Jobs, Veterans]       14
40    41               [CareerArc, Sales]       13
41    42                  [Hiring, Sales]       13
42    43                     [MPLF, 1J1P]       12
43    44               [artwit, iloveart]       12
44    45             [switzerland, swiss]       12
45    46                    [Geneva, Job]       12
46    47             [Photography, Photo]       12
47    48                     [Job, Sales]       12
48    49                    [Jobs, Sales]       12
49    50                      [DJ, DJANE]       12

## Sample_2 

In [453]:
f = open("tweets2.json",'r',encoding='utf-8')
ln2 = 0
sp2_urls = 0
sp2_media = 0
sp2_hash1 = 0
sp2_hash2 = 0
sp2_hashlist1 = []
sp2_hashlist2 = []


for line in f.readlines():
    ln2 += 1
    hlist = []
    dic = json.loads(line)
    
    if dic['urls'] != []:
        sp2_urls += 1
    if dic['media'] != []:
        sp2_media += 1
    if dic['hashtags'] !=[]:
        sp2_hash1 += 1
        for i in dic['hashtags']:
            sp2_hashlist1.append(i['text'])
        if len(dic['hashtags']) >= 2:
            #print(dic['id'])
            for j in dic['hashtags']:
                hlist.append(j['text'])
            sp2_hashlist2.append(hlist)


### Hashtags,Urls,Media

In [454]:
print("Total number of tweets in sample_2: ",ln2 )
print("Total number of tweets contain urls in sample_2: ",sp2_urls,"  pertcentage: ",sp2_urls/ln  )
print("Total number of tweets contain media in sample_2: ",sp2_media,"  pertcentage: ",sp2_media/ln  )
print("Total number of tweets contain at least 1 hashtag in sample_2: ",sp2_hash1,"  pertcentage: ",sp2_hash1/ln  )
print("Total number of tweets contain more than 2 hashtags in sample_2: ",len(sp2_hashlist2),"  pertcentage: ",len(sp2_hashlist2)/ln  )

Total number of tweets in sample_2:  30000
Total number of tweets contain urls in sample_2:  7033   pertcentage:  0.23443333333333333
Total number of tweets contain media in sample_2:  5338   pertcentage:  0.17793333333333333
Total number of tweets contain at least 1 hashtag in sample_2:  7220   pertcentage:  0.24066666666666667
Total number of tweets contain more than 2 hashtags in sample_2:  3342   pertcentage:  0.1114


Total number of tweets in sample_2:  30000

Total number of tweets contain urls in sample_2:  7033   pertcentage:  0.23443333333333333

Total number of tweets contain media in sample_2:  5338   pertcentage:  0.17793333333333333

Total number of tweets contain at least 1 hashtag in sample_2:  7220   pertcentage:  0.24066666666666667

Total number of tweets contain more than 2 hashtags in sample_2:  3342   pertcentage:  0.1114

### The most 50 most frequent hashtags in sample_1:

In [458]:
result = Counter(sp2_hashlist1)
d = sorted(result.items(), key=lambda x: x[1], reverse=True)
df2_1 = pd.DataFrame(d,columns = ["hashtags ","numbers"]).head(50)
ranks = pd.DataFrame(list(range(1,51)),columns=["Rank"]).pop('Rank')
df2_1.insert(0,'Rank',ranks)

This is the most 50 most frequent hashtags in sample_2 and its rank:

In [459]:
df2_1

Rank             hashtags   numbers
0      1           switzerland       95
1      2        CanYouMakeIt16       92
2      3                  quiz       67
3      4        noinonmolliamo       62
4      5         ODproroga2018       61
5      6           Switzerland       59
6      7                   art       45
7      8               NANTAIS       44
8      9                NANTES       43
9     10                Hiring       43
10    11           Controleurs       42
11    12                   Job       42
12    13                  TPMP       41
13    14                zurich       39
14    15              TheVoice       39
15    16                Zurich       38
16    17                   LHC       36
17    18           askbelieber       36
18    19  INCLingapSaTondo2016       36
19    20                  Jobs       34
20    21                Geneva       33
21    22                travel       32
22    23                geneva       31
23    24       ABSCBNFISHSTORY       30
24    25             CareerArc       29
25    26                 swiss       27
26    27                 LNews       27
27    28            TPMPAVegas       27
28    29               TDR2016       26
29    30                 Basel       26
30    31                   CFP       26
31    32         GreenUrbanism       26
32    33                Prince       25
33    34              srfarena       24
34    35                   job       24
35    36              Abstract       24
36    37              Deadline       24
37    38                  May1       24
38    39          bahnbilderde       23
39    40                  alps       21
40    41              lausanne       21
41    42                spring       21
42    43                  lake       21
43    44               Amici15       21
44    45             mountains       20
45    46            ciaodarwin       20
46    47                  love       20
47    48             landscape       20
48    49              Veterans       19
49    50         ABSCBNTWOFACE       19

### The most 50 most frequent hashtag paris in sample_2 

In [465]:
pairs_s2 =[]


for i in sp2_hashlist2:
    
    for m in i:
        for j in i:
            templist = []
            if j != m:
                templist.append(j)
                templist.append(m)
                pairs_s2.append(templist)

pairlist_s2 = []
for i in pairs_s2:
    if (i not in pairlist_s2) & (i[::-1] not in pairlist_s2):
        pairlist_s2.append(i)
count_pair2 =[]
for i in pairlist_s2:
    templist = []
    templist.append(i)
    templist.append(pairs_s2.count(i))
    count_pair2.append(templist)

columnsh = ["hashtags ","numbers"]
count_pdf2 = pd.DataFrame(count_pair2,columns = columnsh)
df2_2 = count_pdf2.sort_values(by="numbers" , ascending=False).head(50)
df2_2.index=[i for i in range(df2_2.shape[0])]


ranks = pd.DataFrame(list(range(1,51)),columns=["Rank"]).pop('Rank')
df2_2.insert(0,'Rank',ranks)

This is the most 50 most frequent hashtags in sample_2and its rank:
            

In [468]:
df2_2

Rank                        hashtags   numbers
0      1  [noinonmolliamo, ODproroga2018]       61
1      2                      [Jobs, Job]       33
2      3                    [Hiring, Job]       26
3      4             [GreenUrbanism, CFP]       26
4      5                   [Hiring, Jobs]       26
5      6              [Hiring, CareerArc]       25
6      7        [Abstract, GreenUrbanism]       24
7      8             [Deadline, Abstract]       24
8      9            [May1, GreenUrbanism]       24
9     10        [Deadline, GreenUrbanism]       24
10    11                 [May1, Abstract]       24
11    12                      [May1, CFP]       24
12    13                  [Deadline, CFP]       24
13    14                  [Abstract, CFP]       24
14    15                 [May1, Deadline]       24
15    16                [CareerArc, Jobs]       19
16    17                 [CareerArc, Job]       19
17    18            [sconti, farmaciauno]       18
18    19                    [Hiring, job]       17
19    20             [promozioni, sconti]       17
20    21        [promozioni, farmaciauno]       17
21    22              [donna, promozioni]       16
22    23             [switzerland, swiss]       16
23    24                  [donna, sconti]       16
24    25             [donna, farmaciauno]       16
25    26               [Hiring, Veterans]       15
26    27                [bellezza, donna]       13
27    28          [bellezza, farmaciauno]       13
28    29             [Photography, Photo]       13
29    30            [travel, switzerland]       13
30    31               [bellezza, sconti]       13
31    32           [bellezza, promozioni]       13
32    33           [sbandy, motormaniaci]       12
33    34                  [sbandy, webtv]       12
34    35            [webtv, motormaniaci]       12
35    36                     [MPLF, 1J1P]       12
36    37                    [artwit, art]       11
37    38            [switzerland, geneva]       11
38    39            [switzerland, zurich]       11
39    40              [instamoto, sbandy]       10
40    41          [motorcycle, instamoto]       10
41    42               [motorcycle, moto]       10
42    43                [instamoto, moto]       10
43    44                  [Job, Veterans]       10
44    45             [moto, motormaniaci]       10
45    46                    [moto, webtv]       10
46    47               [instamoto, webtv]       10
47    48              [motorcycle, webtv]       10
48    49             [motorcycle, sbandy]       10
49    50                   [moto, sbandy]       10

# Discussion part

The unexpect percentage is that there are less tweets with media than contain urls, because I thout Twitter might be used for share some personal interesting things, but if there are more tweets contain urls,it might mean that the twitter is more like a information source rather than a personal statement publication place.

The surprising findings are the hasgtag pairs, from the top 50 hashtag pairs,we can see that the most hashtags that appeared together usually have similar meaning,or have potential relationships, such as "job","hiring".

The reason why they hashtag a tweet with many but similar hashtags may be they want more prople see their tweets, because there are some people who search information by the hashtags. So if you add more hashtags, it will be easier for others to notice or find your tweets.


# My own sample

resample 3000 samples frome the original sample I got

extract the information needed from the big json file

I didn't upload "tweetfrom418.json" since it's too huge ,but I upload the 'tweet3000.json' and 'tweet3001.json', you can run frome next cell if you want.

In [475]:
#this part is resampling 3000 samples frome the data I got
y=list(range(1,54620))
s= random.sample(y,3000) 
f = open("tweetfrom418.json",'r',encoding='utf-8')
ln = 0
for line in f.readlines():
    dic = json.loads(line)
    if ln in s:
        
        with codecs.open('tweet3000.json','a', 'utf-8') as outf:
            #json.dump(dic, outf, ensure_ascii=False)
            
            json.dump({'id': dic['id'], 'entities':dic['entities'] },outf, ensure_ascii=False)
            outf.write('\n')
    ln +=1
f.close()

#this part is extracting the information I need from the resampled json file 
y=list(range(1,54620))
s= random.sample(y,3000) 
f = open("tweet3000.json",'r',encoding='utf-8')
ln = 0
for line in f.readlines():
    dic = json.loads(line)
    with codecs.open('tweet3001.json','a', 'utf-8') as outf:
        if 'media' in dic['entities']:
            json.dump({'id': dic['id'], 'hashtags':dic['entities']['hashtags'] ,'urls':dic['entities']['urls'],'media':dic['entities']['media']},outf, ensure_ascii=False)
        else:
            json.dump({'id': dic['id'], 'hashtags':dic['entities']['hashtags'] ,'urls':dic['entities']['urls'],'media':[]},outf, ensure_ascii=False)
        outf.write('\n')
    ln +=1
f.close()

### run from here

In [477]:
f = open("tweet3001.json",'r',encoding='utf-8')
lnm = 0
spm_urls = 0
spm_media = 0
spm_hash1 = 0
spm_hash2 = 0
spm_hashlist1 = []
spm_hashlist2 = []


for line in f.readlines():
    lnm += 1
    hlist = []
    dic = json.loads(line)
    
    if dic['urls'] != []:
        spm_urls += 1
    if dic['media'] != []:
        spm_media += 1
    if dic['hashtags'] !=[]:
        spm_hash1 += 1
        for i in dic['hashtags']:
            spm_hashlist1.append(i['text'])
        if len(dic['hashtags']) >= 2:
            #print(dic['id'])
            for j in dic['hashtags']:
                hlist.append(j['text'])
            spm_hashlist2.append(hlist)


### Hashtags,Urls,Media  

In [478]:
print("Total number of tweets in my sample: ",lnm )
print("Total number of tweets contain urls in my sample: ",spm_urls,"  pertcentage: ",spm_urls/ln  )
print("Total number of tweets contain media in my sample: ",spm_media,"  pertcentage: ",spm_media/ln  )
print("Total number of tweets contain at least 1 hashtag in my sample: ",spm_hash1,"  pertcentage: ",spm_hash1/ln  )
print("Total number of tweets contain more than 2 hashtags in my sample: ",len(spm_hashlist2),"  pertcentage: ",len(spm_hashlist2)/ln  )

Total number of tweets in my sample:  3000
Total number of tweets contain urls in my sample:  1201   pertcentage:  0.4003333333333333
Total number of tweets contain media in my sample:  373   pertcentage:  0.12433333333333334
Total number of tweets contain at least 1 hashtag in my sample:  501   pertcentage:  0.167
Total number of tweets contain more than 2 hashtags in my sample:  276   pertcentage:  0.092


Total number of tweets in my sample:  3000

Total number of tweets contain urls in my sample:  1201   pertcentage:  0.4003333333333333

Total number of tweets contain media in my sample:  373   pertcentage:  0.12433333333333334

Total number of tweets contain at least 1 hashtag in my sample:  501   pertcentage:  0.167

Total number of tweets contain more than 2 hashtags in my sample:  276   pertcentage:  0.092

### The most 50 most frequent hashtags in my sample 

In [479]:
result = Counter(spm_hashlist1)
d = sorted(result.items(), key=lambda x: x[1], reverse=True)
dfm_1 = pd.DataFrame(d,columns = ["hashtags ","numbers"]).head(50)
ranks = pd.DataFrame(list(range(1,51)),columns=["Rank"]).pop('Rank')
dfm_1.insert(0,'Rank',ranks)

This is the most 50 most frequent hashtags in my sample and their ranks:


In [480]:
dfm_1

Rank            hashtags   numbers
0      1       BBMAsTopSocial       11
1      2          switzerland        8
2      3          Switzerland        8
3      4                 chat        8
4      5                perdu        7
5      6       MarathonNantes        6
6      7           Eurovision        5
7      8               Tatort        5
8      9         architecture        4
9     10               zurich        4
10    11                italy        4
11    12                  ski        4
12    13          photography        4
13    14               travel        4
14    15         lagomaggiore        4
15    16              weather        4
16    17               family        4
17    18             igtravel        3
18    19               italia        3
19    20             chamonix        3
20    21    chamonixmontblanc        3
21    22              mtblanc        3
22    23          montebianco        3
23    24     PatrickSchnidrig        3
24    25  MeinTwitterJubiläum        3
25    26               Brexit        3
26    27                  Ale        3
27    28                 Bier        3
28    29                 Beer        3
29    30                Birra        3
30    31                Bière        3
31    32              Cerveza        3
32    33                 Pivo        3
33    34                   啤酒        3
34    35            Craftbeer        3
35    36            lombardia        3
36    37               Zürich        3
37    38               spring        3
38    39               suisse        3
39    40               sunset        3
40    41                 love        3
41    42              Germany        3
42    43     MusketeersEurope        3
43    44            mountains        3
44    45                Basel        2
45    46                drone        2
46    47         xiaomiredmi5        2
47    48  amazingarchitecture        2
48    49               design        2
49    50                 Dior        2

### The most 50 most frequent hashtag pairs in my sample

In [481]:
pairs_sm =[]


for i in spm_hashlist2:
    
    for m in i:
        for j in i:
            templist = []
            if j != m:
                templist.append(j)
                templist.append(m)
                pairs_sm.append(templist)

pairlist_sm = []
for i in pairs_sm:
    if (i not in pairlist_sm) & (i[::-1] not in pairlist_sm):
        pairlist_sm.append(i)
count_pairm =[]
for i in pairlist_sm:
    templist = []
    templist.append(i)
    templist.append(pairs_sm.count(i))
    count_pairm.append(templist)

columnsh = ["hashtags ","numbers"]
count_pdfm = pd.DataFrame(count_pairm,columns = columnsh)
dfm_2 = count_pdfm.sort_values(by="numbers" , ascending=False).head(50)
dfm_2.index=[i for i in range(df2_2.shape[0])]


ranks = pd.DataFrame(list(range(1,51)),columns=["Rank"]).pop('Rank')
dfm_2.insert(0,'Rank',ranks)

This is the most 50 most frequent hashtag pairs in my sample and their ranks: 

In [484]:
dfm_2

Rank                         hashtags   numbers
0      1                     [perdu, chat]        6
1      2                       [Bier, Ale]        3
2      3                   [Cerveza, Bier]        3
3      4                         [啤酒, Ale]        3
4      5                  [Craftbeer, Ale]        3
5      6                      [Beer, Bier]        3
6      7                     [Birra, Bier]        3
7      8                     [Bière, Bier]        3
8      9                      [Pivo, Bier]        3
9     10                    [Cerveza, Ale]        3
10    11                        [啤酒, Bier]        3
11    12                 [Craftbeer, Bier]        3
12    13                     [Birra, Beer]        3
13    14                     [Bière, Beer]        3
14    15                   [Cerveza, Beer]        3
15    16                      [Pivo, Beer]        3
16    17                       [Pivo, Ale]        3
17    18                      [Bière, Ale]        3
18    19                    [Bière, Birra]        3
19    20                      [Birra, Ale]        3
20    21                       [Beer, Ale]        3
21    22                   [chamonix, ski]        3
22    23          [chamonixmontblanc, ski]        3
23    24                    [mtblanc, ski]        3
24    25                [montebianco, ski]        3
25    26     [chamonixmontblanc, chamonix]        3
26    27               [mtblanc, chamonix]        3
27    28           [montebianco, chamonix]        3
28    29      [mtblanc, chamonixmontblanc]        3
29    30  [montebianco, chamonixmontblanc]        3
30    31            [montebianco, mtblanc]        3
31    32   [Switzerland, PatrickSchnidrig]        3
32    33                [Germany, weather]        3
33    34                 [Craftbeer, Beer]        3
34    35                        [啤酒, Beer]        3
35    36                  [Cerveza, Birra]        3
36    37                   [Pivo, Cerveza]        3
37    38                   [italy, italia]        3
38    39                     [Pivo, Birra]        3
39    40                   [Craftbeer, 啤酒]        3
40    41                        [啤酒, Pivo]        3
41    42              [Craftbeer, Cerveza]        3
42    43                     [啤酒, Cerveza]        3
43    44                 [Craftbeer, Pivo]        3
44    45                [Craftbeer, Bière]        3
45    46                       [啤酒, Birra]        3
46    47                     [Pivo, Bière]        3
47    48                  [Cerveza, Bière]        3
48    49                [Craftbeer, Birra]        3
49    50                       [啤酒, Bière]        3

## comparison between samples 

### between sample_1 and sample_2 

#### top 20 

In [500]:
inter201_2 = list(set(list(df1_1.head(20)['hashtags '])).intersection(set(list(df2_1.head(20)['hashtags ']))))
union201_2 = list(set(list(df1_1.head(20)['hashtags '])).union(set(list(df2_1.head(20)['hashtags ']))))
print("the number of common hashtags in sample_1 and sample_2",len(inter201_2))
print("the ration of common hashtags in sample_1 and sample_2",len(inter201_2)/len(union201_2))

the number of common hashtags in sample_1 and sample_2 17
the ration of common hashtags in sample_1 and sample_2 0.7391304347826086


#### top 50  

In [502]:
inter501_2 = list(set(list(df1_1['hashtags '])).intersection(set(list(df2_1['hashtags ']))))
union501_2 = list(set(list(df1_1['hashtags '])).union(set(list(df2_1['hashtags ']))))
print("the number of common hashtags in sample_1 and sample_2",len(inter501_2))
print("the ration of common hashtags in sample_1 and sample_2",len(inter501_2)/len(union501_2))

the number of common hashtags in sample_1 and sample_2 38
the ration of common hashtags in sample_1 and sample_2 0.6129032258064516


### between sample_1 and my sample



#### top 20 

In [508]:
inter201_m = list(set(list(df1_1.head(20)['hashtags '])).intersection(set(list(dfm_1.head(20)['hashtags ']))))
union201_m = list(set(list(df1_1.head(20)['hashtags '])).union(set(list(dfm_1.head(20)['hashtags ']))))
print("the number of common hashtags in sample_1 and my sample",len(inter201_m))
print("the ration of common hashtags in sample_1 and my sample",len(inter201_m)/len(union201_m))

the number of common hashtags in sample_1 and my sample 3
the ration of common hashtags in sample_1 and my sample 0.08108108108108109


#### top 50

In [509]:
inter501_m = list(set(list(df1_1['hashtags '])).intersection(set(list(dfm_1['hashtags ']))))
union501_m = list(set(list(df1_1['hashtags '])).union(set(list(dfm_1['hashtags ']))))
print("the number of common hashtags in my sample and sample_1",len(inter501_m))
print("the ration of common hashtags in my sample and sample_1",len(inter501_m)/len(union501_m))

the number of common hashtags in my sample and sample_1 9
the ration of common hashtags in my sample and sample_1 0.0989010989010989


### between sample_1 and sample_2



#### top 20 

In [506]:
inter202_m = list(set(list(df1_1.head(20)['hashtags '])).intersection(set(list(dfm_1.head(20)['hashtags ']))))
union202_m = list(set(list(df1_1.head(20)['hashtags '])).union(set(list(dfm_1.head(20)['hashtags ']))))
print("the number of common hashtags in my sample and sample_2",len(inter202_m))
print("the ration of common hashtags in my sample and sample_2",len(inter202_m)/len(union202_m))

the number of common hashtags in my sample and sample_2 3
the ration of common hashtags in my sample and sample_2 0.08108108108108109


#### top 50 

In [510]:
inter502_m = list(set(list(df1_1['hashtags '])).intersection(set(list(dfm_1['hashtags ']))))
union502_m = list(set(list(df1_1['hashtags '])).union(set(list(dfm_1['hashtags ']))))
print("the number of common hashtags in my sample and sample_2",len(inter502_m))
print("the ration of common hashtags in my sample and sample_2",len(inter502_m)/len(union502_m))

the number of common hashtags in my sample and sample_2 9
the ration of common hashtags in my sample and sample_2 0.0989010989010989


# discussion

The most interesting I noticed is that there are so many hashtags about beer,in so many different languages.Maybe because there are a beer festival here in April, so in order to do the commercial promotion, they will public more tweets with hashtags about beer.

Another thing interesting is that there is really little hashtags can be related with Notre Dame, I thout there will be more hashtags about Notre Dame bacause of the fire, but no,there are not so many. only some hashtags about architecture can be related to the Notre Dame.This may because that I collect the data from 18th April,but I expected there are more hashtags about the fire.

There is only 3 top 20 hashtags in commen between hashtags I collected and the sample_1 or sample_2. And most of these common hashtags are hashtags about the location or country such as "Switzerland","Zurich",etc.

There more common hasgtags between sample_1 and sample_2,so that may indicate that the thing people foucused have changed.The only thing that didn't change with time  is the hashtags about the country.That may because that Switzerland is a tourising country, there were a lot of people spent there holiday here,so when tourist get here, they will publish tweets with hashtags about Swiss.



